core/src/main/scala/mongo4cats/client/MongoClient.scala

```diff
@@ -16,55 +16,76 @@
 
 package mongo4cats.client
 
-import cats.effect.{Async, Resource, Sync}
-import cats.syntax.flatMap._
+import cats.~>
+import cats.arrow.FunctionK
+import cats.effect.{Async, Resource}
+import cats.implicits._
+import fs2.Stream
 import com.mongodb.connection.ClusterDescription
 import com.mongodb.reactivestreams.client.{MongoClient => JMongoClient, MongoClients}
-import mongo4cats.bson.Document
 import mongo4cats.database.MongoDatabase
 import mongo4cats.helpers._
+import org.bson.Document
 
 import scala.jdk.CollectionConverters._
 
-abstract class MongoClient[F[_]] {
-  def clusterDescription: ClusterDescription
+trait MongoClient[F[_]] {
+  def clusterDescription: F[ClusterDescription]
   def getDatabase(name: String): F[MongoDatabase[F]]
-  def listDatabaseNames: F[Iterable[String]]
-  def listDatabases: F[Iterable[Document]]
-  def listDatabases(session: ClientSession[F]): F[Iterable[Document]]
-  def startSession(options: ClientSessionOptions): F[ClientSession[F]]
-  def startSession: F[ClientSession[F]] = startSession(ClientSessionOptions.apply())
+  def listDatabaseNames: Stream[F, String]
+  def listDatabases(
+      clientSession: Option[ClientSession[F]] = None
+  ): Stream[F, Document]
```

Comment:
I think this should return `BsonDocument`, so we avoid `Document` in any of the signatures.

Reply:
Yeah, but this will be `map((x: Document) => BsonDocument.parse(x.toJson())` since `Document` is what driver returns.

Reply:
Actually, ignore my comment, it might be `listDatabase(session, classOf[BsonDocument])`

Reply:
Or even `.map(_.toBsonDocument)` since `Document <: Bson`?